File: Upcoder/Lập trình nâng cao/7.Đệ quy/Số cuối cùng/LASTNUM.py
# Hàm tính tổng các chữ số của một số
def digit_sum(num):
    return sum(int(c) for c in num)

# Hàm tính "last number" (số cuối cùng)
def last_num(num):
    while len(num) > 1:
        num = str(digit_sum(num)) # Chuyển lại số thành chuỗi để tiếp tục xử lý
    return int(num[0]) # Trả về số cuối cùng

# Đọc đầu vào và xuất kết quả
n = input()
print(last_num(n))

File: Upcoder/Lập trình nâng cao/7.Đệ quy/Số cuối cùng/LASTNUM.cpp
#include <bits/stdc++.h>
using namespace std;

// Hàm tính tổng các chữ số của một số
int digitSum(string num) {
    int sum = 0;
    for (char c : num) {
        sum += c - '0'; // Chuyển ký tự thành số nguyên
    }
    return sum;
}

// Hàm tính "last number" (số cuối cùng) theo phương pháp số học
int LastNum(string num) {
    while (num.length() > 1) {
        int sum = digitSum(num);
        num = to_string(sum); // Chuyển lại số thành chuỗi để tiếp tục xử lý
    }
    return num[0] - '0'; // Trả về số cuối cùng
}

int main() {
    string n;
    cin >> n;
    cout << LastNum(n) << endl;
    return 0;
}
